File: src/main/java/com/example/transactions/application/dto/StatisticResponse.java
package com.example.transactions.application.dto;

import com.example.transactions.domain.Statistic;
import lombok.Value;

@Value
public class StatisticResponse {
    public Double sum;
    public Double avg;
    public Double max;
    public Double min;
    public Integer count;

    public static StatisticResponse from(Statistic statistic) {
        return new StatisticResponse(statistic.sum, statistic.avg, statistic.max, statistic.min, statistic.count);
    }
}

File: application/src/main/kotlin/com/example/application/dto/StatisticResponse.kt
package com.example.application.dto

import com.example.domain.Statistic

data class StatisticResponse(val sum: Double, val avg: Double, val max: Double, val min: Double, val count: Int) {
    companion object {
        operator fun invoke(statistic: Statistic): StatisticResponse {
            return StatisticResponse(statistic.sum, statistic.avg, statistic.max, statistic.min, statistic.count)
        }
    }
}
